### PR TITLE
add a way to authenticate as a github app

### PR DESCRIPTION
**Pull request by TomasTomecek**
https://github.com/TomasTomecek/ansible-bender/issues/12
https://github.com/TomasTomecek/ansible-bender/pull/13

* [x] The auth stuff is not right, I need to play with it a bit more.

### CODE REVIEW

**Comment by TomasTomecek** [comment]
I can see that older versions of this PR passed testing so bumping everything to F29 should be okay. That's what I'm using actually and it works just fine.

File: Makefile [hunk 1]
```diff
@@ -20,7 +20,7 @@ test-in-container: image-test
 	docker run -it -e GITHUB_USER=${GITHUB_USER} -e GITHUB_TOKEN=${GITHUB_TOKEN} $(TEST_IMAGE_NAME)
 
 test:
-	PYTHONPATH=$(CURDIR) pytest -v
+	PYTHONPATH=$(CURDIR) pytest-3 -v
```
Comment (jpopelka) [inline Makefile]: Not going to fly. In [Dockerfile.test](https://github.com/user-cont/release-bot/blob/master/Dockerfile.test#L16) we install pytest with pip3, so `pytest --version` should return 3.x.y.
`pytest-3` is a name of binary in `python3-pytest` rpm, which we don't use

But please add the `--showlocals`

Comment (TomasTomecek) [reply Makefile]: Figured it out finally locally. Will add.

**Comment by TomasTomecek** [comment]
I went through the codacy report and most of the issues are style. I can fix some if needed.

**Comment by TomasTomecek** [comment]
Okay, I was able to release ansible-bender to github and pypi with code base present in this PR. Feel free to do the final review.

**Comment by TomasTomecek** [comment]
AAAAAaaaaa!

File: release_bot/github.py [hunk 2]
```diff
@@ -13,16 +13,82 @@
...
+        return r
+
+
+class GitHubApp:
+    def __init__(self, app_id, private_key_path):
+        self.session = requests.Session()
+        self.session.headers.update(dict(
+            accept='application/vnd.github.machine-man-preview+json'))
+        self.private_key = None
+        self.private_key_path = private_key_path
```
Comment (webknjaz) [inline release_bot/github.py]: @TomasTomecek You can actually save it in openshift secrets accessible via env var :)

Comment (TomasTomecek) [reply release_bot/github.py]: Yep, we can do that. I'm running the pod as rootless podman container now, so for me it's good enough to have the path stored in a config. But for future, we should utilize openshift secrets, that's a good point.

Comment (webknjaz) [reply release_bot/github.py]: That's what I did in `ansiwatch-bot`, while `check-in` still supports loading the key from a file.

**Pull request merged**